Loop/Managers/VersionCheckServicesManager.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  VersionCheckServicesManager.swift
+//  Loop
+//
+//  Created by Rick Pasetto on 9/8/21.
+//  Copyright © 2021 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import LoopKit
+
+final class VersionCheckServicesManager {
+
+    private lazy var log = DiagnosticLog(category: "VersionCheckServicesManager")
+
+    private lazy var dispatchQueue = DispatchQueue(label: "com.loopkit.Loop.VersionCheckServicesManager")
+    
+    private var versionCheckServices = Locked<[VersionCheckService]>([])
+    
+    init() {}
+
+    func addService(_ versionCheckService: VersionCheckService) {
+        versionCheckServices.mutate { $0.append(versionCheckService) }
+    }
+
+    func restoreService(_ versionCheckService: VersionCheckService) {
+        versionCheckServices.mutate { $0.append(versionCheckService) }
+    }
+
+    func removeService(_ versionCheckService: VersionCheckService) {
+        versionCheckServices.mutate { $0.removeAll { $0.serviceIdentifier == versionCheckService.serviceIdentifier } }
+    }
+    
+    func checkVersion(currentVersion: String) -> VersionUpdate {
+        let semaphore = DispatchSemaphore(value: 0)
+        var results = [String: Result<VersionUpdate, Error>]()
+        let services = versionCheckServices.value
+        services.forEach { versionCheckService in
+            dispatchQueue.async {
+                versionCheckService.checkVersion(bundleIdentifier: Bundle.main.bundleIdentifier!, currentVersion: currentVersion) { result in
+                    self.dispatchQueue.async {
+                        results[versionCheckService.serviceIdentifier] = result
+                        semaphore.signal()
+                    }
+                }
+            }
+            semaphore.wait()
+        }
+        
+        var aggregatedVersionUpdate = VersionUpdate.noneNeeded
+        results.forEach { key, value in
+            switch value {
+            case .failure(let error):
+                self.log.error("Error from version check service %{public}@: %{public}@", key, error.localizedDescription)
```

Comment:
See note in LoopKit PR about handling a version check service that doesn't know about a particular bundle identifier. It isn't really an error.

Reply:
Also, how do we handle the following scenario:
- version check service reports that an update is needed
- user does NOT update
- later we retry the version check and the version check service **encounters a network error**
- the version check service reports an error, but how does the version check service manager resolve that?
- it cannot simply report that no update is needed (because that isn't true)
- thoughts?

Reply:
That's a good corner case...a bit out of scope for this PR, but I'm imagining that it should be the responsibility of the `VersionCheckService` implementation to handle that.  Namely: it should realize this and respond with the last known `VersionUpdate`.  Thoughts?

Reply:
Okay, then I guess that needs to be implemented (or a separate Jira ticket created) to do that in TidepoolService.

Reply:
Working on that now :)